File: Sistema de Gestao Academico/Web.Server/Models/Usuario.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Web.Server.Models
{
    public class Usuario
    {
        public int Id { get; set; }
        [Required]
        public string Nome { get; set; }
        [Required]
        public string Sobrenome { get; set; }
        [Required]
        public string Cpf { get; set; }
        [Required]
        public DateTime DataNascimento { get; set; }
        [Required]
        public string Email { get; set; }
        [Required]
        public Cargos Cargo { get; set; }
        [Required]
        public string Senha { get; set; }

        [NotMapped]
        public string Token { get; set; }
    }
}

File: Sistema de Gestao Academico/Web/ClientApp/src/app/_models/usuario.ts
import { Cargos } from "../_enums/cargos.enum";

export class Usuario {
  id: string;
  cpf: string;
  nome: string;
  sobrenome: string;
  dataNascimento: Date;
  email: string;
  senha: string;
  cargo: Cargos;
  token: string;

  get nomeCompleto(): string {
    return `${this.nome} ${this.sobrenome}`;
  }

  static map(usuario: Usuario): Usuario {
    let u = null;
    if (usuario) {
      u = new Usuario()
      u.id = usuario.id;
      u.cpf = usuario.cpf;
      u.nome = usuario.nome;
      u.sobrenome = usuario.sobrenome;
      u.email = usuario.email;
      u.senha = usuario.senha;
      u.cargo = usuario.cargo;
    }
    return u;
  }
}
